kore/src/Kore/Syntax/Pattern.hs

```diff
@@ -105,6 +107,11 @@ newtype
     deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
     deriving anyclass (Debug, Diff)
 
+type FutuPattern variable t =
```

Comment:
We should revert these changes.

Reply:
This is not the data structure that you originally introduced (I think that's gone). This is a new helper data structure that I defined and that is in use in the `External` module.
I suppose it could be moved to that module though, would that be good?